lib/jira/commands/new.rb

```diff
@@ -44,16 +44,18 @@ def run
       attr_accessor :ticket
 
       def params
-        {
+        parms = {
           fields: {
             project:     { id: project['id'] },
             issuetype:   { id: issue_type['id'] },
             summary:     summary,
             description: description,
-            parent:      @parent.nil? ? {} : { key: @parent },
-            components:  @components.nil? ? [] : @components
+            parent:      @parent.nil? ? nil : { key: @parent },
+            components:  @components.nil? ? nil : @components
           }
         }
+        parms[:fields].delete_if{|k,v| v.nil? || v.eql?([])}
```

Comment:
instead of introducing the `parms` local, can we move the delete_if as part of the hash declaration?
```
def params
  {
    fields: {
      ...
    }.delete_if{ |_,v| v.nil? || v == [] }
  }
end